jcl/src/java.base/share/classes/openj9/internal/tools/attach/target/AttachHandler.java

```diff
@@ -323,6 +331,87 @@ private boolean initialize() throws IOException {
+				final String lastAccessTime = "lastAccessTime"; //$NON-NLS-1$
+				final String commonDir = CommonDirectory.getCommonDirFileObject().getPath();
+				fileAccessTimeUpdaterThread = new Thread("Attach API update file access time") { //$NON-NLS-1$
+					@Override
+					public void run() {
+						VM.markCurrentThreadAsSystem();
+
+						long lastMillis = System.currentTimeMillis();
+						long sleepMillis = targetIntervalMillis;
+						for (;;) {
+							try {
+								Thread.sleep(sleepMillis);
+							} catch (InterruptedException ie) {
+								// ignore this non-fatal exception
+								IPC.logMessage("fileAccessTimeUpdaterThread received an InterruptedException: " + ie.getMessage()); //$NON-NLS-1$
+							}
+							if (isAttachApiTerminated()) {
+								IPC.logMessage("AttachAPI was terminated, fileAccessTimeUpdaterThread exits"); //$NON-NLS-1$
+								break;
+							}
+							long currentMillis = System.currentTimeMillis();
+							long passedMillis = currentMillis - lastMillis;
+							if (passedMillis >= sleepMillis) {
+								FileTime fileTime = FileTime.fromMillis(currentMillis);
+								IPC.logMessage("fileAccessTimeUpdaterThread currentMillis = " + currentMillis); //$NON-NLS-1$
+								try {
+									// Common directory : _attachlock
+									File attachLock = new File(commonDir, CommonDirectory.ATTACH_LOCK);
+									if (attachLock.exists()) {
+										// _attachlock only appears when there was an attaching request
+										Files.setAttribute(Paths.get(attachLock.getPath()), lastAccessTime, fileTime);
+										IPC.logMessage("fileAccessTimeUpdaterThread updated access time = " + attachLock); //$NON-NLS-1$
+									}
+									// Common directory : _controller
+									Files.setAttribute(Paths.get(commonDir, CommonDirectory.CONTROLLER_LOCKFILE), lastAccessTime, fileTime);
+									// Common directory : _notifier
+									Files.setAttribute(Paths.get(commonDir, CommonDirectory.CONTROLLER_NOTIFIER), lastAccessTime, fileTime);
+									// Target directory : attachNotificationSync
+									Files.setAttribute(Paths.get(TargetDirectory.getSyncFileObject().getPath()), lastAccessTime, fileTime);
+									// Target directory : attachInfo
+									Files.setAttribute(Paths.get(TargetDirectory.getAdvertisementFileObject().getPath()), lastAccessTime, fileTime);
+								} catch (IOException ioe) {
+									// ignore this non-fatal exception
+									IPC.logMessage("fileAccessTimeUpdaterThread received an IOException : " + ioe.getMessage()); //$NON-NLS-1$
+								}
+								// reset for next update
+								sleepMillis = targetIntervalMillis;
+								lastMillis = currentMillis;
+							} else {
+								// continue sleep()
+								sleepMillis = targetIntervalMillis - passedMillis;
+								IPC.logMessage("fileAccessTimeUpdaterThread currentMillis = " + currentMillis //$NON-NLS-1$
+										+ ", passedMillis = " + passedMillis + ", sleepMillis = " + sleepMillis); //$NON-NLS-1$ //$NON-NLS-2$
```

Comment:
I suggest this is easier to read with one value per line:
```java
								IPC.logMessage("fileAccessTimeUpdaterThread currentMillis = " + currentMillis //$NON-NLS-1$
										+ ", passedMillis = " + passedMillis //$NON-NLS-1$
										+ ", sleepMillis = " + sleepMillis); //$NON-NLS-1$
```

Reply:
Updated